crates/forge/src/running.rs

```diff
@@ -157,3 +165,32 @@ pub(crate) fn run_from_test_case(
         Err(err) => Err(err.into()),
     }
 }
+
+fn extract_fork_when_id_passed(
+    predefined_forks: Option<&Vec<PredefinedFork>>,
+    fork_alias: &str,
+) -> Option<ForkStateReader> {
+    let fork = predefined_forks?
+        .iter()
+        .find(|fork| fork.name == fork_alias);
+
+    let block_id = fork?
+        .block_id
+        .iter()
+        .map(|(id_type, value)| match id_type.as_str() {
+            "number" => Some(BlockId::Number(value.parse().unwrap())),
+            "hash" => Some(BlockId::Hash(value.to_field_element())),
+            "tag" => match value.as_str() {
+                "Latest" => Some(BlockId::Tag(BlockTag::Latest)),
+                "Pending" => Some(BlockId::Tag(BlockTag::Pending)),
+                _ => None,
+            },
+            _ => None,
```

Comment:
Those should be `unreachable!()` after we do validation in `crates/forge/src/scarb.rs`